### PR TITLE
feat(scroll direction): emit scroll direction events

### PR DESCRIPTION
**Pull request by jackhkmatthews**
emit direction up or down event on resize or scroll

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,15 +60,15 @@ A working example can be found inside [/src](https://github.com/thisissoon/angul
 ### Scroll direction
 
 ```html
-<nav class="foo" snScrollCollapse>
+<nav class="foo" snScrollCollapse (scrollDirectionEvents)="scrollDirectionHandler($event)">
```
Comment (edoparearyee) [inline README.md]: can you change the name of the event to `scrollDirectionChange`, it should be like other event names such as `click`, `mouseEnter` etc...

File: src/app/scroll-collapse/scroll-collapse.directive.spec.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { ElementRef, SimpleChanges } from '@angular/core';
 import { WindowRef } from '@thisissoon/angular-inviewport';
 
 import { ScrollCollapseDirective } from './scroll-collapse.directive';
+import { Direction } from '../../../dist';
```
Comment (edoparearyee) [inline src/app/scroll-collapse/scroll-collapse.directive.spec.ts]: This should be imported from `src/app/scroll-collapse/...` etc...

File: src/app/scroll-collapse/scroll-collapse.directive.ts [hunk 3]
```diff
@@ -93,7 +101,7 @@ export class ScrollCollapseDirective implements AfterViewInit, OnDestroy {
    */
   @HostBinding(classes.directionUpClass)
   public get isScrollingUp(): boolean {
-    return this.scrollDirection === Direction.UP;
+    return this.lastScrollDirection === Direction.UP;
```
Comment (edoparearyee) [inline src/app/scroll-collapse/scroll-collapse.directive.ts]: what's the logic behind the name change?

**Review by edoparearyee — CHANGES_REQUESTED** [review]

File: src/app/scroll-collapse/scroll-collapse.directive.ts (hunk 3 above)
Comment (jackhkmatthews) [reply src/app/scroll-collapse/scroll-collapse.directive.ts]: Given the introduction of a new 'scrollDirection' variable name I thought the added specifity would be helpful.

However given the above requested change to 'scrollDirectionChange' I think this is no longer required.

File: README.md (hunk 1 above)
Comment (jackhkmatthews) [reply README.md]: Updated

**Review by edoparearyee — APPROVED** [review]

**Pull request merged**